src/Map/MapRasterizer_P.cpp

```diff
@@ -521,32 +544,78 @@ void OsmAnd::MapRasterizer_P::rasterizePolyline(
     }
     else
     {
+        float hmargin = 0.0f;
+        auto drawShifted = [&, this](float hmargin)
+        {
+            SkPath pathHmargin;
+            if (calcPathByTrajectory(context, points31, pathHmargin, hmargin))
+                canvas.drawPath(pathHmargin, paint);
+        };
         if (updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_minus2, false))
-            canvas.drawPath(path, paint);
+        {
```

Comment:
Changing the code structure doesn't look good and produces too much of new code.
Suggested approach
```
if (updatePaint(context, paint, *refToPath*, primitive->evaluationResult, PaintValuesSet::Layer_minus2, false)) 
   canvas.drawPath(refToPath, paint);

```

So in that sense updatePaint -> updatePathPaint will hide complexity inside if and won't be copy pasted

Reply:
It was moved to separate method.
BTW: `hmargin` -- it's about geometry modification, but `updatePaint()` - it's about configuration of the paint, brush etc. Thus it shouldn't be take together - logically - IMHO.